### PR TITLE
Invalid summary input

### PR DESCRIPTION
**Pull request by jakesabathia2**
fix https://github.com/apple/turicreate/issues/2871

### CODE REVIEW

File: src/python/turicreate/toolkits/_model.py [hunk 1]
```diff
@@ -514,17 +514,17 @@ def summary(self, output=None):
...
-            return _toolkit_serialize_summary_struct( self, \
+            return _toolkit_serialize_summary_struct(self, \
                                             *self._get_summary_struct() )
-
-        try:
-            print(self.__repr__())
-        except:
-            return self.__class__.__name__
+        else:
+            raise ToolkitError("Unsupported argument " + str(output) + " for summary.")
```
Comment (TobyRoseman) [inline src/python/turicreate/toolkits/_model.py]: For error message, I like to make it very clear it's referring to a parameter. I would change: `summary.` to `\"summary\" parameter.`

File: src/python/turicreate/toolkits/_model.py [hunk 2]
```diff
@@ -675,16 +675,17 @@ def summary(self, output=None):
...
         elif output == 'dict':
-            return _toolkit_serialize_summary_struct( self, \
+            return _toolkit_serialize_summary_struct(self, \
                                             *self._get_summary_struct() )
-        try:
-            print(self.__repr__())
-        except:
-            return self.__class__.__name__
+        else:
+            raise ToolkitError("Unsupported argument " + str(output) + " for summary.")
```
Comment (TobyRoseman) [inline src/python/turicreate/toolkits/_model.py]: same as previous comment

File: src/python/turicreate/toolkits/_model.py (hunk 1 above)
Comment (jakesabathia2) [reply src/python/turicreate/toolkits/_model.py]: good!

File: src/python/turicreate/toolkits/_model.py (hunk 2 above)
Comment (jakesabathia2) [reply src/python/turicreate/toolkits/_model.py]: good!

**Pull request merged**